### ISSUE TITLE
Incorrect ASHRAE climate zone application

### ISSUE THREAD

**Issue by DavidGoldwasser** [NREL/OpenStudio#3148]
I was running an existing workflow in 2.5.1 and hit odd issue with Change Building Location measure when I apply ASHRAE climate zone. It inputs the value for institution 1 but no description, and then CEC description for institution 2 but without a value. I then confirmed the same behavior in the prototype measure. Below is the object for small office 2a

```
OS:ClimateZones,
  {94c3875c-3601-4ae2-a595-92e37d440c9d}, !- Handle
  ,                                       !- Active Institution
  ,                                       !- Active Year
  ,                                       !- Climate Zone Institution Name 1
  ,                                       !- Climate Zone Document Name 1
  ,                                       !- Climate Zone Document Year 1
  2A,                                     !- Climate Zone Value 1
  CEC,                                    !- Climate Zone Institution Name 2
  California Climate Zone Descriptions,   !- Climate Zone Document Name 2
  1995,                                   !- Climate Zone Document Year 2
  ;                                       !- Climate Zone Value 2

```

Oddly, the GUI shows the ASHRAE climate zone as 2A (maybe it hard codes first institution as ASHRAE?), but this is breaking downstream measures that apply constructions based on ASHRAE climate zones.

**Comment by DavidGoldwasser** [comment]
For some additional information, when I launch OS app and make a new model and save before doing anything else, it is already populated with CEC institution information and looks like this.

OS:ClimateZones,
  {4d7fb5cb-394a-42e6-8380-def9794fd6ed}, !- Handle
  ,                                       !- Active Institution
  ,                                       !- Active Year
  ,                                       !- Climate Zone Institution Name 1
  ,                                       !- Climate Zone Document Name 1
  ,                                       !- Climate Zone Document Year 1
  ,                                       !- Climate Zone Value 1
  CEC,                                    !- Climate Zone Institution Name 2
  California Climate Zone Descriptions,   !- Climate Zone Document Name 2
  1995,                                   !- Climate Zone Document Year 2
  ;                                       !- Climate Zone Value 2

When I use the GUI to set the ASHRAE climate zone I see the same behavior as the measures.

OS:ClimateZones,
  {4d7fb5cb-394a-42e6-8380-def9794fd6ed}, !- Handle
  ,                                       !- Active Institution
  ,                                       !- Active Year
  ,                                       !- Climate Zone Institution Name 1
  ,                                       !- Climate Zone Document Name 1
  ,                                       !- Climate Zone Document Year 1
  5B,                                     !- Climate Zone Value 1
  CEC,                                    !- Climate Zone Institution Name 2
  California Climate Zone Descriptions,   !- Climate Zone Document Name 2
  1995,                                   !- Climate Zone Document Year 2
  ;                                       !- Climate Zone Value 2

cc @asparke2

**Comment by macumber** [comment]
Relevant lines:

https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/ClimateZones.cpp#L182

https://github.com/NREL/OpenStudio-measures/blob/develop/nrel_published/ChangeBuildingLocation/measure.rb#L232

**Comment by DavidGoldwasser** [comment]
For reference, the line after I set climate zone works and returns the ASHRAE value.

```
runner.registerInfo("Setting Climate Zone to #{climateZones.getClimateZones('ASHRAE').first.value}")
```

Here is code in standards called by downstream create typical model measure for climate zone for construction lookup. The empty CEC may be the issue I'm hitting?
https://github.com/NREL/openstudio-standards/blob/5fe0f175d9156c567dfe83a4c4c9948bb65e616c/lib/openstudio-standards/standards/Standards.Model.rb#L3005-L3019

**Comment by DavidGoldwasser** [comment]
Solution is to add line 'climateZones.clear' before climateZones.set is used. This does two things.
1. It makes sure there are not more than one ASHRAE climate zones which would make the set method not work
2. It removes any empty climate zone objects like CEC that is in new OSM in app, and was throwing off climate zone lookup in openstudio-standards.

I will lower severity, but will leave this open. I see a few actions related to this, so I was going to leave it open for now.

1. Remove the empty CEC climate zone from the new OSM made in OS app. If a users sets ASHRAE climate zone manually they don't get the benefit of the `climateZones.clear` used in the measure, and as a result current OS standards code would still choose empty CEC climate zone.
2. openstudio-standards could be updated in lookup to not choose climate zone with blank value
3. Update measure writing guide to recommend that climateZones.append not be used. This is to avoid more than one climate zone with the same name which breaks API methods to lookup climate zone.
4. Update other measure that set climate zones, maybe look at measure that get climate zones to see that they are more robust ,but I think that is a lower priority.

**Comment by jmarrec** [comment]
@DavidGoldwasser Can you check now that https://github.com/NREL/OpenStudio/pull/3871 was merged in?

**Comment by DavidGoldwasser** [comment]
I don't have build environment setup, but I can check when I have an installer.

**Comment by jmarrec** [comment]
Rcs should be rolled out soon anyways, but if you have time to test, here's current develop: 

* Windows: https://openstudio-ci-builds.s3.amazonaws.com/develop/OpenStudio-3.0.0-beta%2Bd05bc3a9ba-Windows.exe

* Mac: https://openstudio-ci-builds.s3.amazonaws.com/develop/OpenStudio-3.0.0-beta%2Bd05bc3a9ba-Darwin.dmg

**Comment by jmarrec** [comment]
Fixed via #3871
